Adding all dependent libraries to the Stan Math Library

syclik opened the pull request:
#### Summary:

Includes copies of:

- Boost
- Eigen
- Google Test
- CppLint

to the source distribution under the `lib/` folder, like in Stan.

#### Intended Effect:

Allows the Stan Math Library to be complete with its dependencies.

#### How to Verify:

Run any test.

#### Side Effects:

Stan will remove its copy of the libraries.

#### Documentation:

Updated in `README.md`. Will need to update the wiki for testing instructions also.

#### Reviewer Suggestions: 

anyone.

syclik commented:
@bob-carpenter, mind taking a look at the `licenses` folder? If you're ok with that, we should be good to go with this pull request.

bob-carpenter commented:
I moved `licenses/math-license.txt` to `licenses/stan-math-library-license.txt`.

I also rewrote the license to clarify that we're not trying to assert we own the code and Columbia doesn't.

```
Copyright (c) 2011--2015, Stan Developers or their Assignees
```

syclik commented:
thanks! I'll merge.

On Thu, Jul 30, 2015 at 3:11 PM, Bob Carpenter <notifications@github.com>
wrote:

> I moved licenses/math-license.txt to
> licenses/stan-math-library-license.txt.
>
> I also rewrote the license to clarify that we're not trying to assert we
> own the code and Columbia doesn't.
>
> Copyright (c) 2011--2015, Stan Developers or their Assignees
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stan-dev/math/pull/132#issuecomment-126442662>.
>
